Add support for browsersync configuration in external rc file

cgmartin opened the pull request:
Addresses #1 

This PR adds support for an external configuration file `bs-config.js` (or `bs-config.json`) that matches [Browser Sync's config format](https://www.browsersync.io/docs/options/). 

You can run `browser-sync init` in the angular2-tour-of-heroes repo to generate a sample config file (must `npm i -g browser-sync` beforehand).

*Most* `lite-server` command-line arguments will override any configuration file settings. All except for the `--indexFile` argument, which will not be applied if the configuration file has explicit middleware defined. We can fix this if it is a problem.

The README still needs to be updated, if this change is agreed upon.

cgmartin commented:
An alternative solution is to deprecate `lite-server` in the `angular2-tour-of-heroes` project, and use `browser-sync` directly. See details here: https://github.com/johnpapa/angular2-tour-of-heroes/pull/51

johnpapa commented:
@cgmartin Let's track our ideas in an issue and propose either a way to expose BS entirely or use a fully custom script that deprecates lite-server.

cgmartin commented:
Issue discussion here: https://github.com/johnpapa/lite-server/issues/22

cgmartin commented:
Updated PR based on feedback in #22 

Simplified `lite-server.js` entrypoint to merge separate `config-defaults.js` file (browser-sync config format) with user's custom, optional `bs-config.js` file.

Updated README with example configuration overrides. This should address #23, #12, #24, #6.

Fixed request logger format to skip printing of "(Unknown)" from #13.
